test/test-retaining-ajax-calls.js

```diff
@@ -0,0 +1,163 @@
+(function(qunit, $) {
+  'use strict';
+
+  var t = qunit.test;
+
+  /* ---------------------------------- */
+  qunit.module( 'Retaining Ajax Calls' );
+  /* ---------------------------------- */
+
+  t('Setting defaults', function(assert) {
+    assert.equal($.mockjaxSettings.retainAjaxCalls, true, '$.mockjaxSettings.retainAjaxCalls defaults to true');
+  });
+
+  t('Mocked calls are retained when the setting is set to true', function(assert) {
```

Comment:
This is already tested in the core test file. Let's move all tests of the unmocked and mocked arrays into this file (and out of core) to be consistent.